src/CodeCoverage/Collector.php

```diff
@@ -13,6 +13,11 @@
  */
 class Collector
 {
+	const
```

Comment:
There is some common code, so you would need three classes – one for each strategy and one base class. The logic for deciding which strategy to use would probably end-up extracted outside of those classes in \Tester\Environment::setupCoverage.

To summarize – it would require more code but should be possible.